libcodechecker/analyze/store_handler.py

```diff
@@ -340,44 +340,76 @@ def addReport(storage_session,
         # large databases?
         reports = session.query(Report) \
             .filter(Report.bug_id == bug_hash, Report.run_id == run_id)
-        # Check for duplicates by bug hash.
-        LOG.debug("checking duplicates")
+
+        # Detection status for the newly stored reports.
+        new_report_status = 'new'
+
+        # Report id for a duplicate report, if it was already stored in
+        # the database. Bug hash and report path should match.
+        duplicate_report_id = None
+
+        if reports:
+            LOG.debug("There are reports with the same hash.")
+
         for report in reports:
-            LOG.debug("there is a possible duplicate")
-            # TODO: file_id and path equality check won't be necessary
-            # when path hash is added.
-            if report.checker_id == checker_id and \
-                    is_same_event_path(report.id, events, session):
 
-                new_status = None
+            detection_status_for_hash = \
+                    storage_session.get_detection_status(run_id,
+                                                         report.bug_id)
+            if not detection_status_for_hash:
+                # Calculate the detection status for a report
+                # not processed in the current storage session yet.
+                if report.detection_status == 'new':
+                    report.detection_status = 'unresolved'
+                    report.file_id = file_id
+
+                elif report.detection_status == 'unresolved':
+                    report.detection_status = 'unresolved'
```

Comment:
You don't have to set the `detection_status` field to `unresolved` because it's already `unresolved` in this block.